juju/client/connection.py

```diff
@@ -656,7 +656,7 @@ async def rpc(self, msg, encoder=None):
                 # if it is triggered by the pinger, then this RPC call will
                 # be cancelled when the pinger is cancelled by the reconnect,
                 # and we don't want the reconnect to be aborted halfway through
-                await jasyncio.wait([self.reconnect()])
+                await jasyncio.wait([jasyncio.create_task(self.reconnect())])
```

Comment:
How is this different from just `await self.reconnect()`? It looks like `asyncio.wait()` is supposed to be used when you're waiting for multiple concurrent tasks. I'm not at all up on `asyncio` -- maybe @dimaqq could chime in too?

Reply:
True, I just assumed that `asyncio` was being used for a good reason, but I'm not super familiar with it either. Perhaps related to the comment above this line?
```python
# the reconnect has to be done in a separate task because,
# if it is triggered by the pinger, then this RPC call will
# be cancelled when the pinger is cancelled by the reconnect,
# and we don't want the reconnect to be aborted halfway through
```

Reply:
This comment and call have been introduced in https://github.com/juju/python-libjuju/pull/148/files.

From the comment, we can infer that the caller wants the coroutine `reconnect` to be continue even if the `rpc` method is itself cancelled.

One of the side effect of the previous implementation of `asyncio.wait` was shielding the coroutine from the caller cancellation.

To make this behaviour clearer, we could instead:
```python3
await jasyncio.shield(asyncio.create_task(self.reconnect()))
```

The shield would still yield a CancelledError, but the task inside would continue in the background.

Reply:
Given that there's a single item in the wait list, wdyt about this instead?

```suggestion
                await self.reconnect()
```